src/utils.rs

```diff
@@ -10,6 +11,55 @@ pub unsafe fn debug_reset_static_data() {
     ffi::cairo_debug_reset_static_data()
 }
 
+pub fn status_to_result<T>(status: ffi::cairo_status_t, obj: T) -> Result<T, Error> {
```

Comment:
If you let it take a `FnOnce() -> T` closure instead then you could also use it in all the other places where an actual value is returned

Reply:
I have found only one place where it can be useful which is the `add_outline` 
```
pub fn add_outline(
        &self,
        parent_id: i32,
        name: &str,
        link_attribs: &str,
        flags: PdfOutline,
    ) -> Result<i32, Error> {
        let name = CString::new(name).unwrap();
        let link_attribs = CString::new(link_attribs).unwrap();

        let res = unsafe {
            ffi::cairo_pdf_surface_add_outline(
                self.0.to_raw_none(),
                parent_id,
                name.as_ptr(),
                link_attribs.as_ptr(),
                flags.bits() as _,
            ) as _
        };

        self.status()?;
        Ok(res)
    }
```
The rest of the cases returns a `Result<(), Error>`. I can still change it to something that takes a FnOnce if that will make the code more idiomatic

Reply:
In that case leave it but remove the `obj` parameter and always make it return a `Result<(), Error>` :)